internal/services/cdn/cdn_endpoint_custom_domain_resource_test.go

```diff
@@ -30,20 +32,49 @@ func preCheck(t *testing.T) {
 	}
 }
 
+func preCheckUserManagedCertificate(t *testing.T) {
+	preCheck(t)
+	variables := []string{
+		// PFX File
+		"ARM_TEST_DNS_CERTIFICATE",
+
+		// Subdomain Name, this needs to be the same value as set to "CN" in the certificate.
+		"ARM_TEST_DNS_SUBDOMAIN_NAME",
+	}
```

Comment:
by moving this function onto the `CdnEndpointCustomDomainResource` struct we can check the variables rather than the env variables here (which may not be used below?)